### ISSUE TITLE
Commute history.

### ISSUE THREAD

**Issue by hemanth** [sindresorhus/module-requests#19]
This is more of `daemon` module.

The idea:

Say I commute between `pointA` and `pointB` everyday  (on road) at different points of time, but i'm interested in get the traffic behaviour for a particular day at a particular time or find the best time to commute given a day and time range, based on the historical data, yes there live data available, but traffic varies drastically from weekdays to weekends or even public holidays, it would be nice to know the best time to commute, yes it might not be accurate, but it will give us a rough estimate.  

Well once we have the data, we can play around.

P.S: Tried looking into google maps API, they don't really expose an open API for the same.

**Comment by sindresorhus** [comment]
Google Maps have a pretty accurate travel estimate, based on history, time of day, real-time traffic info, and a lot of advanced algorithms.

**Comment by hemanth** [comment]
Yes, I agree with that, but there is no way for us to get the data on how the was traffic for a given route on a particular day in the history.

**Comment by dcondrey** [comment]
http://trafficpredict.com/

**Comment by Qix-** [comment]
Hi, Uber employee here.

This is much harder than you're making it out to be. There aren't going to be readily-available datasets that are accurate enough to get updated times like you're wanting - unless you want to source them yourself, get a partnership with a company like Waze (like Google does) or Google themselves.

For example, Uber had to acquire a substantial portion of Bing just to get most of the assets necessary to have our own real-time traffic analysis systems. I'm not on any of those teams but believe me when I say this isn't a small feat, even for a single one-off trip.

As well, sourcing good data is expensive and requires many users using a single app to collect recent traffic conditions and then process all of those hundreds (thousands?) of terabytes a day of information to infer what the traffic conditions might actually be (rush hour vs. accident vs. weather, etc.) so don't be surprised if companies are reluctant to give that kind of information up.

While you won't have to write to-scale systems to achieve this, it is definitely a big data problem.

Your _best bet_ is to try to find an API for a reputable mapping system, like Google, Apple Maps, or Bing maps (Waze might even supply some of their data if you [look hard enough](https://www.waze.com/about/dev)). Keep in mind Google _does_ seem to factor in traffic conditions in a lot of their results* so perhaps that API isn't as hopeless as it seems.

<sub><sub>However, I am not, nor have I ever been, a Google engineer so this is based on experience, not facts.</sub></sub>

Something else to take from experience: historical traffic patterns mean very little. Traffic engineering and city planning projects are ongoing (seriously, construction sucks just about everywhere) so historical data loses its relevance pretty quickly.

Hope this helps.

**Comment by hemanth** [comment]
Thanks for the detailed response, Google Map shows this option to select different timings, I wonder if that's just an estimate based on historical data, but not able to find any open APIs though. 

<img width="408" alt="screen shot 2016-10-11 at 7 30 54 pm" src="https://cloud.githubusercontent.com/assets/18315/19273208/43c2e4d6-8fe9-11e6-9d0d-32bc2921b3ea.png">
